Cassiopee/Generator/Generator/IBM.py

```diff
@@ -1082,6 +1081,56 @@ def buildOctree(tb, dimPb=3, vmin=15, snears=0.01, snearFactor=1., dfars=10., df
 #==============================================================================
 # 
 #==============================================================================
+def createRefinementBodies(tb, dimPb=3, hmod=0.01):
+    """Creates refinement bodies from the immersed boundaries to extend the finest resolution in the fluid domain."""
+    import Geom.IBM as D_IBM
+    import Geom.Offset as O
+
+    pointsPerUnitLength = 10 if dimPb == 3 else 100
+
+    refinementBodies = []
+
+    tb = Internal.rmNodesFromName(tb, "SYM")
+    tb = Internal.rmNodesFromName(tb, "*_sym")
+
+    snears    = Internal.getNodesFromName(tb, 'snear')
+    h         = min(snears, key=lambda x: x[1])[1][0]*4.
+
+    for z in Internal.getZones(tb):
+        snear = Internal.getNodeFromName(z, 'snear')[1]
+        zname = z[0]
+        if snear <= 1.5*h/4.:
```

Comment:
I might be missing something. why 1.5*h/4 specifically?

Reply:
That's because I forgot to change the declaration of h in my last commit. Initially, it was also the definition of the constant step in the Cartesian grid that I used to calculate the offset... As for the 1.5, it's simply a tolerance.